vendor/views/vendor.py

```diff
@@ -166,14 +158,14 @@ def post(self, request, *args, **kwargs):
         return redirect('vendor:cart')      # Redirect to cart on success
 
 
-class AccountInformationView(LoginRequiredMixin, TemplateView):
+class AccountInformationView(LoginRequiredMixin, SetSiteToRequestMixin, TemplateView):
     template_name = 'vendor/checkout.html'
 
     def get(self, request, *args, **kwargs):
         context = super().get_context_data(**kwargs)
         clear_session_purchase_data(request)
         
-        invoice = get_purchase_invoice(request.user)
+        invoice = get_purchase_invoice(request.user, request.site)
```

Comment:
use the new function here to get the site object instead of modifying the request object.

Reply:
changed to: `invoice = get_purchase_invoice(request.user, get_site_from_request(request))`